source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -525,6 +525,33 @@ TEST(NiFakeServiceTests, NiFakeService_BoolArrayOutputFunction_CallsBoolArrayOut
   EXPECT_THAT(response.an_array(), ElementsAreArray(expected_response_booleans, number_of_elements));
 }
 
+TEST(NiFakeServiceTests, NiFakeService_BoolArrayInputFunction_CallsBoolArrayInputFunction)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViInt32 number_of_elements = 3;
+  ViBoolean expected_array[] = { VI_FALSE, VI_TRUE, VI_TRUE };
+  EXPECT_CALL(library, BoolArrayInputFunction(kTestViSession, number_of_elements, _))
+    .With(Args<2, 1>(ElementsAreArray(expected_array)))
+    .WillOnce(Return(kDriverSuccess));
+
+  ::grpc::ServerContext context;
+  nifake_grpc::BoolArrayInputFunctionRequest request;
+  request.mutable_vi()->set_id(session_id);
+  request.set_number_of_elements(number_of_elements);
+  bool input_array[] = { false, true, true };
+  for (bool item : input_array) {
+    request.add_an_array(item);
+  }
```

Comment:
It's just as many lines and simpler to just call `add_an_array` multiple times:
```suggestion
    request.add_an_array(false);
    request.add_an_array(true);
    request.add_an_array(true);
```